main.tf

```diff
@@ -226,6 +250,18 @@ resource "aws_route" "public_internet_gateway" {
   }
 }
 
+resource "aws_route" "public_firewall" {
+  count = var.create_vpc && var.enable_firewall ? length(var.public_subnets) : 0
+
+  route_table_id         = element(aws_route_table.public.*.id, count.index)
+  destination_cidr_block = "0.0.0.0/0"
+  vpc_endpoint_id        = element([for ss in tolist(aws_networkfirewall_firewall.this[0].firewall_status[0].sync_states) : ss.attachment[0].endpoint_id if ss.attachment[0].subnet_id == aws_subnet.firewall[count.index].id], 0)
```

Comment:
Because of this, it requires to specify at least as many firewall_subnets as public_subnets. 

Should we default to first firewall_subnet in case aws_subnet.firewall[count.index] doesn't exist?

Reply:
Hello, are u able to share a terraform example of your configuration has issues?

I think i understand your first sentence, but not sure how defaulting to first subnet helps.

Reply:
Hi @ryanoolala, sure - this is a simple example that fails:
```
  cidr = "10.10.0.0/18"

  azs             = ["us-east-1a", "us-east-1b", "us-east-1c"]
  public_subnets  = ["10.10.4.0/24", "10.10.5.0/24"]
  private_subnets = ["10.10.8.0/21", "10.10.16.0/21", "10.10.24.0/21"]

  enable_firewall      = true
  firewall_subnets     = ["10.10.1.0/28"]
  firewall_policy_arn  = dependency.firewall-policy.outputs.arn

  single_nat_gateway = true
  enable_nat_gateway = true
```

will lead to 
```
╷
│ Error: Invalid index
│
│   on main.tf line 258, in resource "aws_route" "public_firewall":
│  258:   vpc_endpoint_id        = element([for ss in tolist(aws_networkfirewall_firewall.this[0].firewall_status[0].sync_states) : ss.attachment[0].endpoint_id if ss.attachment[0].subnet_id == aws_subnet.firewall[count.index].id], 0)
│     ├────────────────
│     │ aws_subnet.firewall is tuple with 1 element
│     │ count.index is 1
│
│ The given key does not identify an element in this collection value: the
│ given index is greater than or equal to the length of the collection.
```

Reply:
Hello, in this case you have only 1 firewall endpoint in `us-east-1a`, public in 1a,1b and private in 1a,1b,1c

Is there a reason why you have your setup in such a way, where firewall is not highly available, while the private is the most highly available one? because if 1a goes down with your firewall, you will not have any path to internet anymore, and typically if you are already using multi az setups, you would want to maintain that across scenarios.

Reply:
Indeed, that's more an artificial example. But I think it could be a valid use case to use more public subnets than firewall subnets. E.g. this

```
  azs             = ["us-east-1a", "us-east-1b", "us-east-1c"]
  public_subnets  = ["10.10.4.0/24", "10.10.5.0/24", "10.10.6.0/24", "10.10.7.0/24", "10.10.8.0/24"]
  private_subnets = ["10.10.8.0/21", "10.10.16.0/21", "10.10.24.0/21"]
  firewall_subnets     = ["10.10.1.0/28", "10.10.2.0/28", "10.10.3.0/28"]
```

So this could be a setup for certain users and it wouldn't make sense to have more than three firewall subnets here.

Reply:
hmm yes i think it possible to have more public subnets, I'm also thinking if its possible to have more firewall subnets than the number of AZs in general, which i don't feel it is good to have even though it does help in cost savings, but defeats the purpose of having multi availability zones.

My current thoughts is that the firewall should support scenarios where
`length(var.public_subnets) > length(var.firewall_subnets) && length(var.azs) == length(var.firewall_subnets)`

I'll try to find some time to work on this

Reply:
Just to let you know @ryanoolala there's a team here in the UK really appreciating the work you're doing on this feature 🥇

Reply:
Hello @wiegandf, I've made the changes to support additional public subnets, let me know if it works for you?

@deploymentking Thank you for the appreciation! I'm glad to contribute to the community.

Reply:
This functionality would be a godsend for us! :) Currently manually reworking a bunch of code to make it work :thinking: